### PR TITLE
[FLINK-25742][akka] Remove the serialization of rpc invocation at Fli…

### PR DESCRIPTION
**Pull request by KarmaGYZ**
…nk side.

## What is the purpose of the change

Remove the serialization of rpc invocation at Flink side.

## Brief change log

Remove the redundant serialization in `RemoteRpcInvocation`.

## Verifying this change

This change is already covered by existing tests, such as `MessageSerializationTest`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Review by xintongsong — APPROVED** [review]
LGTM. +1 for merging.

**Comment by dawidwys** [comment]
Sorry for nitpicking. Could we please pay more attention to PRs description? We should serve as an example if we require contributors to fill it in.

**Comment by KarmaGYZ** [comment]
@dawidwys I'll fill that.

**Comment by dawidwys** [comment]
Thanks @KarmaGYZ ! I know from personal experience it might be daunting at times :sweat_smile:

File: flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/messages/RemoteRpcInvocation.java [hunk 1]
```diff
@@ -35,13 +34,12 @@
  * contains a non-serializable object, then an {@link IOException} is thrown.
  */
 public class RemoteRpcInvocation implements RpcInvocation, Serializable {
```
Comment (zentol) [inline flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/messages/RemoteRpcInvocation.java:36]: javadocs are outdated

File: flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/MessageSerializationTest.java [hunk 2]
```diff
@@ -139,32 +139,6 @@ public void testSerializableRemoteMessageTransfer() throws Exception {
         assertThat(linkedBlockingQueue.take(), Is.<Object>is(expected));
     }
 
-    /**
-     * Tests that a message which exceeds the maximum frame size is detected and a corresponding
-     * exception is thrown.
-     */
-    @Test(expected = IOException.class)
-    public void testMaximumFramesizeRemoteMessageTransfer() throws Exception {
```
Comment (zentol) [inline flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/MessageSerializationTest.java:147]: Which tests now covers the case of oversized messages?

**Review by zentol — COMMENTED** [review]
Can you explain what problem you're trying to solve?

**Comment by KarmaGYZ** [comment]
@zentol This PR is an improvement of the job execution for Flink OLAP. In our internal benchmark, the serialization of the `RemoteRpcInvocation` at the Flink side will increase the end-to-end latency of a 1000*1000 `WordCount` for about 100ms.

File: flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/MessageSerializationTest.java (hunk 2 above)
Comment (KarmaGYZ) [reply flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/MessageSerializationTest.java:147]: Before this PR, Flink will eagerly check the oversized messages. Now, when there is an oversized message, the job will fail with Akka's `AskTimeoutException` and there will be an error log from Akka which indicates the exact size of that message. As Flink does not take the responsibility to detect oversized messages, I think we don't need such tests anymore.

**Comment by zentol** [comment]
> Without this serialization

But things are serialized _anyway_ because of Akka; it just happens later. So where is this performance gain coming from?

File: flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/MessageSerializationTest.java (hunk 2 above)
Comment (zentol) [reply flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/MessageSerializationTest.java:147]: There should still be a test that checks the behavior if this error case occurs. E.g., that the response future is completed exceptionally.

**Comment by KarmaGYZ** [comment]
@zentol Now, the `RemoteRpcInvocation` are serialized twice, first in Flink and second in Akka. This PR removes the first serialization in Flink.

**Comment by KarmaGYZ** [comment]
@flinkbot run azure

**Comment by KarmaGYZ** [comment]
@flinkbot run azure

**Review by zentol — APPROVED** [review]

**Comment by KarmaGYZ** [comment]
@flinkbot run azure

**Comment by KarmaGYZ** [comment]
@flinkbot run azure

**Pull request closed without merging**